From c0f42ed759b403c785faf2b519f8d79ea6b0ad7e Mon Sep 17 00:00:00 2001
From: Adrien Destugues <pulkomandy@pulkomandy.tk>
Date: Sun, 13 Mar 2022 13:09:45 +0100
Subject: Fix build under Haiku

- We already define umode_t, don't redefine it.
- We have an header file named image.h which gets included instead of
  the uboot one, use a more specific path.

diff --git a/arch/arm/include/asm/types.h b/arch/arm/include/asm/types.h
index 900b261..0b7b3fa 100644
--- a/arch/arm/include/asm/types.h
+++ b/arch/arm/include/asm/types.h
@@ -3,7 +3,7 @@
 
 #include <asm-generic/int-ll64.h>
 
-typedef unsigned short umode_t;
+//typedef unsigned short umode_t;
 
 /*
  * These aren't exported outside the kernel to avoid name space clashes
diff --git a/include/u-boot/ecdsa.h b/include/u-boot/ecdsa.h
index 6e0269e..c6bda3f 100644
--- a/include/u-boot/ecdsa.h
+++ b/include/u-boot/ecdsa.h
@@ -7,7 +7,7 @@
 #define _ECDSA_H
 
 #include <errno.h>
-#include <image.h>
+#include "../include/image.h"
 #include <linux/kconfig.h>
 
 /**
diff --git a/lib/hash-checksum.c b/lib/hash-checksum.c
index 8f2a42f..cf99255 100644
--- a/lib/hash-checksum.c
+++ b/lib/hash-checksum.c
@@ -14,7 +14,7 @@
 #include "fdt_host.h"
 #endif
 #include <hash.h>
-#include <image.h>
+#include "../include/image.h"
 
 int hash_calculate(const char *name,
 		    const struct image_region *region,
diff --git a/tools/imagetool.h b/tools/imagetool.h
index 5169b02..4ab9f29 100644
--- a/tools/imagetool.h
+++ b/tools/imagetool.h
@@ -21,7 +21,7 @@
 #include <unistd.h>
 #include <u-boot/sha1.h>
 
-#include <image.h>
+#include "../include/image.h"
 
 #include "fdt_host.h"
 
-- 
2.30.2


From 30cee83e2807f1469a00376d751841b75e20e19c Mon Sep 17 00:00:00 2001
From: Adrien Destugues <pulkomandy@pulkomandy.tk>
Date: Sun, 13 Mar 2022 15:04:42 +0100
Subject: Quirk to ignore invalid video mode

My TV has an invalid video mode in its EDID data, using it results in
incorrect timings and a black screen. Just ignore that mode (with a
pretty crude detection).

diff --git a/drivers/video/sunxi/sunxi_display.c b/drivers/video/sunxi/sunxi_display.c
index 5a21f7a..757cd74 100644
--- a/drivers/video/sunxi/sunxi_display.c
+++ b/drivers/video/sunxi/sunxi_display.c
@@ -289,6 +289,9 @@ static int sunxi_hdmi_edid_get_mode(struct sunxi_display_priv *sunxi_display,
 
 	/* Take the first usable detailed timing */
 	for (i = 0; i < 4; i++, t++) {
+		/* Ignore bogus EDID mode we can't handle */
+		if (EDID_DETAILED_TIMING_VERTICAL_ACTIVE(*t) == 540 && EDID_DETAILED_TIMING_HORIZONTAL_ACTIVE(*t) == 1920)
+			continue;
 		r = video_edid_dtd_to_ctfb_res_modes(t, mode);
 		if (r == 0)
 			break;
-- 
2.30.2

